File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function() {
  var self = this;

  self.complements = {
    'G': 'C',
    'C': 'G',
    'T': 'A',
    'A': 'U'
  };

  self.isValidInput = function(dna) {
    dna.split('').forEach(function(x) {
      if(Object.keys(self.complements).indexOf(x) === -1) throw new Error('Invalid input');
    });
    return true;
  };

  self.toRna = function(dna) {
    if(self.isValidInput(dna)) return dna.split('').map(function(x) {
      return self.complements[x];
    }).join('');
  };
};

module.exports = DnaTranscriber;

File: python/rna-transcription/rna_transcription.py
dict = {
'G': 'C',
'C': 'G',
'T': 'A',
'A': 'U'
}
valids = 'GCTA'
def to_rna(dna):
    s = []
    for char in dna:
        if char in valids:
            s.append(dict[char])
    return ''.join(s) if len(s) == len(dna) else ''
